gslib/utils/copy_helper.py

```diff
@@ -3681,17 +3681,16 @@ def _WriteRowToManifestFile(self, url):
     """Writes a manifest entry to the manifest file for the url argument."""
     row_item = self.items[url]
```

Comment:
What does `row_item` contain at this point -- always a unicode object, or a version-dependent type (bytes in PY2, unicode in PY3)?

It would be neat if we could ensure the former, and would prevent a lot of the crazy casting/{en,de}coding work going on below.

Reply:
It's a variety of types, including version-dependent strings, ints, and datetime tuples, which is why I tried `str()` on all the items when it was a list comprehension.

```
{'description': '', 'bytes': 4247, 'size': 4247, 'end_time': datetime.datetime(2019, 4, 8, 23, 39, 8, 87305), 'source_uri': 'file:///usr/local/google/home/cball/.vimrc', 'result': 'OK', 'start_time': datetime.datetime(2019, 4, 8, 23, 39, 7, 749583), 'md5': 'YSH3vG6cGl7y4+hRBf1G7w==', 'destination_uri': 'gs://do-not-delete-cball-test/.vimrc'}
```

Reply:
> I think we're better off reading stuff from the file so that the contents we get are unicode (using io.open), since we know the file's format will be utf-8 (being that we control the manifest file).

Ok, changed it to use io.open